File: csharp/1401.circle-and-rectangle-overlapping.cs
/*
 * @lc app=leetcode id=1401 lang=csharp
 *
 * [1401] Circle and Rectangle Overlapping
 */

using System;

// @lc code=start
public class Solution {
    public bool CheckOverlap(int radius, int x_center, int y_center, int x1, int y1, int x2, int y2) {
        int cx = Math.Max(x1, Math.Min(x2, x_center)),
            cy = Math.Max(y1, Math.Min(y2, y_center)),
            dx = x_center - cx,
            dy = y_center - cy;
        return radius * radius >= dx * dx + dy * dy;
    }
}
// @lc code=end

File: java/1401.circle-and-rectangle-overlapping.java
/*
 * @lc app=leetcode id=1401 lang=java
 *
 * [1401] Circle and Rectangle Overlapping
 */

// @lc code=start
class Solution {
    public boolean checkOverlap(int radius, int xCenter, int yCenter, int x1, int y1, int x2, int y2) {
        x1 -= xCenter;
        x2 -= xCenter;
        y1 -= yCenter;
        y2 -= yCenter;
        return radius * radius >=
            (
                x1 * x2 > 0
                ? Math.min(x1 * x1, x2 * x2)
                : 0
            )
            +
            (
                y1 * y2 > 0
                ? Math.min(y1 * y1, y2 * y2)
                : 0
            );
    }
}
// @lc code=end

